tests/test_unit/test_visualization.py

```diff
@@ -0,0 +1,107 @@
+import os
+from unittest.mock import MagicMock, patch
+
+import cv2
+import numpy as np
+import pytest
+import torch
+
+from crabs.detection_tracking.visualization import (
+    draw_bbox,
+    draw_detection,
+    save_images_with_boxes,
+)
+
+
+def test_draw_bbox():
+    image = np.zeros((100, 100, 3), dtype=np.uint8)
+    draw_bbox(image, (10, 10), (20, 20), (0, 255, 0))
+    cv2.imwrite("bbox_test.png", image)
```

Comment:
It is a bit odd that the image is saved to the current working directory - seems like a good idea to use [pytest's `tmp_path` fixture](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html). It creates a temporary directory unique to the test invocation, and pytest handles cleanup for us after testing.